R-package/src/Makevars.in

```diff
@@ -0,0 +1,56 @@
+CXX_STD = CXX11
+
+PKGROOT=.
+
+LGB_CPPFLAGS = \
+    @LGB_CPPFLAGS@ \
+    -DUSE_SOCKET \
+    -DLGB_R_BUILD
+
+CPICFLAGS = -fPIC
+
+PKG_CPPFLAGS = \
+    -I$(PKGROOT)/include \
+    $(LGB_CPPFLAGS)
+
+PKG_CXXFLAGS = \
+    @OPENMP_CXXFLAGS@ \
+    -pthread
+
+PKG_LIBS = \
+    @OPENMP_CXXFLAGS@ \
+    @OPENMP_LIB@ \
+    -pthread
+
+OBJECTS = \
```

Comment:
Investigate `wildcard` alternative (mostly a note for myself): https://github.com/microsoft/LightGBM/pull/3188#discussion_r449240610.

Reply:
@StrikerRUS @jameslamb FYI, you may consider how XGBoost deals with multiple sources in the R package. We use an approach called amalgamation (also known as unity build), where all the C++ sources get included in a single C++ source file: https://github.com/dmlc/xgboost/blob/master/amalgamation/xgboost-all0.cc. This way, the Makevars file is kept simple: https://github.com/dmlc/xgboost/blob/master/R-package/src/Makevars.in.

Reply:
@hcho3 Hmm, interesting solution! Thanks a lot for sharing your experience!

@jameslamb I suppose we can adopt that approach a little bit later to not slow down merging of this PR. WDYT?

Reply:
oooooooo thanks @hcho3! Yes I think this is an interesting idea but not necessary for this PR. I documented it in #3249